Assembler.UnitTests/MiddleMessageHandlerTests.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using Assembler.Base;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+using Moq;
+using NUnit.Framework;
+
+namespace Assembler.UnitTests
+{
+    [TestFixture]
+    public class MiddleMessageHandlerTests
+    {
+        private Mock<ITimeBasedCache<BaseMessageInAssembly>> _cacheMock;
+        private Mock<IFactory<BaseFrame, string>> _identifierFactoryMock;
+        private Mock<IMessageEnricher<BaseFrame, BaseMessageInAssembly>> _enricherMock;
+        private Mock<ICreator<BaseMessageInAssembly>> _messageInAssemblyCreatorMock;
+
+        private List<BaseMessageInAssembly> _assembledMessages;
+        private string _identifierString;
+
+        private MiddleMessageHandler<BaseFrame, BaseMessageInAssembly> _handler;
+
+        [SetUp]
+        public void Setup()
+        {
+            _cacheMock = new Mock<ITimeBasedCache<BaseMessageInAssembly>>();
+            _enricherMock = new Mock<IMessageEnricher<BaseFrame, BaseMessageInAssembly>>();
+            _messageInAssemblyCreatorMock = new Mock<ICreator<BaseMessageInAssembly>>();
+            _identifierString = Utilities.GetIdentifierString();
+            _identifierFactoryMock = Utilities.GetIdentifierMock();
+
+            _assembledMessages = new List<BaseMessageInAssembly>();
+
+            _handler = new MiddleMessageHandler<BaseFrame, BaseMessageInAssembly>(_cacheMock.Object,
+                _identifierFactoryMock.Object, _messageInAssemblyCreatorMock.Object, _enricherMock.Object,
+                Utilities.GetLoggerFactory());
+
+            _handler.MessageAssemblyFinished += _assembledMessages.Add;
+        }
+
+        [TearDown]
+        public void Teardown()
+        {
+            _identifierFactoryMock.VerifyNoOtherCalls();
+            _enricherMock.VerifyNoOtherCalls();
+            _cacheMock.VerifyNoOtherCalls();
+            _messageInAssemblyCreatorMock.VerifyNoOtherCalls();
+        }
+
+        [Test]
+        public void Handle_IdentifierThrowsException_FrameNotBeingUsed()
+        {
+            // Arrange
+            var frame = new Mock<BaseFrame>(FrameType.Initial);
```

Comment:
Not exactly critical, but perhaps you would like to change `FrameType.Initial` to `FrameType.Middle`, considering that would be the actual value

Reply:
Also appears in lines 73, 106

Reply:
You are right.

Reply:
Well,now it's `FrameType.End` when I would expect it to be `FrameType.Middle`. Again, this doesn't bother me this much because these are tests, but just thought I'd let you know

Reply:
I changed the Middle one and then went to change the End one and ended up changing the middle again....
Check it now.

Reply:
👍